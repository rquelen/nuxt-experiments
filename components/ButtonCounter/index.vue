<template>
  <button @click="incrementCounter" class="test">Button data: {{ counter }}</button>
</template>

<script>
export default {
  name: 'ButtonCounter',
  data () {
    return {
      counter: 0,
    };
  },
  methods: {
    incrementCounter: function () {
      this.counter += 1;
      this.$emit('increment');
    },
  },
};
</script>

<style scoped>
  .test {
    font-size: 30px;
  }
</style>
