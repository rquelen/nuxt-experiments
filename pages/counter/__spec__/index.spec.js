import {shallow} from '@vue/test-utils';
import ButtonCounter from '~/components/ButtonCounter';

import Counter from '../index';

describe('/counter', () => {
  it('should match snapshot', () => {
    // when
    const wrapper = shallow(Counter);

    // then
    expect(wrapper.element).toBeDefined();
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('asyncData', () => {
    it('should initialize data with API value asynchronously', async () => {
      // given
      const $axios = {
        get: jest.fn(),
      };

      $axios.get.mockReturnValue(Promise.resolve({ data: '11' }));

      const wrapper = shallow(Counter);

      // when
      const asyncData = await wrapper.vm.$options.asyncData({app: {$axios}});

      // then
      expect($axios.get).toHaveBeenCalledWith('http://www.mocky.io/v2/5a01affc300000da45fac0cf');
      expect(asyncData).toEqual({ total: '11' });
    });
  });

  it('should initialize counter to 0', () => {
    // when
    const wrapper = shallow(Counter);

    // then
    const paragraph = wrapper.find('p');
    expect(paragraph.text()).toContain('0');
  });

  it('should increment counter on event increment from either buttons', () => {
    // given
    const wrapper = shallow(Counter);
    const buttonCounters = wrapper.findAll(ButtonCounter);

    // when
    buttonCounters.at(0).vm.$emit('increment');
    buttonCounters.at(1).vm.$emit('increment');
    wrapper.update();

    // then
    const paragraph = wrapper.find('p');
    expect(paragraph.text()).toContain('2');
  });
});
