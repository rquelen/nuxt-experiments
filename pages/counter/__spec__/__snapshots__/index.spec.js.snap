// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/counter should match snapshot 1`] = `
<div
  class="container"
>
  <p
    class="text"
  >
    Page data: 0
  </p>
   
  <div
    class="buttons-wrapper"
  >
    <!---->
     
    <!---->
  </div>
   
  <div>
    <nuxt-link
      exact=""
      to="/"
    >
      Home
    </nuxt-link>
  </div>
</div>
`;
